test/burn/org/apache/cassandra/concurrent/LongSharedExecutorPoolTest.java

```diff
@@ -135,19 +139,27 @@ private void testPromptnessOfExecution(long intervalNanos, float loadIncrement)
         {
             if (System.nanoTime() > until)
             {
-                System.out.println(String.format("Completed %.0fK batches with %.1fM events", runs * 0.001f, events * 0.000001f));
+                logger.info(String.format("Completed %.0fK batches with %.1fM events", runs * 0.001f, events * 0.000001f));
                 events = 0;
                 until = System.nanoTime() + intervalNanos;
                 multiplier += loadIncrement;
-                System.out.println(String.format("Running for %ds with load multiplier %.1f", TimeUnit.NANOSECONDS.toSeconds(intervalNanos), multiplier));
+                logger.info(String.format("Running for %ds with load multiplier %.1f", TimeUnit.NANOSECONDS.toSeconds(intervalNanos), multiplier));
             }
 
             // wait a random amount of time so we submit new tasks in various stages of
             long timeout;
+            boolean timeoutIsMax = false;
             if (pending.isEmpty()) timeout = 0;
-            else if (Math.random() > 0.98) timeout = Long.MAX_VALUE;
+            else if (Math.random() > 0.98)
+            {
+                timeout = System.nanoTime() + TimeUnit.HOURS.toNanos(1);
+                timeoutIsMax = true;
+            }
             else if (pending.size() == executorCount) timeout = pending.first().timeout;
-            else timeout = (long) (Math.random() * pending.last().timeout);
+            else timeout = System.nanoTime() + (long) (Math.random() * (pending.last().timeout - System.nanoTime()));
+
+            if ((timeout - System.nanoTime()) < 0)
+                logger.warn("Timeout set to a negative value {}ms", TimeUnit.NANOSECONDS.toMillis(timeout - System.nanoTime()));
```

Comment:
minor:
calling `System.nanoTime())` every here at now can lead to fancy results, maybe we should store it in some variable? Technically it can be different in condition and log msg ;-)